internal/pkg/api/server/templates/http/error.go

```diff
@@ -118,19 +118,19 @@ func errorHandler() func(context.Context, http.ResponseWriter, error) {
 }
 
 // errorFormatter returns a function that adds HTTP status code to the given error.
-func errorFormatter() func(err error) goaHTTP.Statuser {
+func errorFormatter() func(ctx context.Context, err error) goaHTTP.Statuser {
```

Comment:
@webrouse tak bych asi potřeboval aspoň trochu navést. Vyšla Goa 3.9 a podle Slacku je tam breaking change v tom error formatteru, má nový parametr s kontextem. (Tenhle PR: https://github.com/goadesign/goa/pull/3137)
Upravil jsem to jak mi přišlo správný (ten kontext se ve `formatError()` asi nevyužije, stejně jako doteď v `handleError()`), ale failují E2E testy na tom že user chyby nejsou správně detekovaný a vrací se místo nich 500. Nemůžu přijít na to kde ten problém je. 😕 

<img width="994" alt="image" src="https://user-images.githubusercontent.com/215660/193536079-e5065b43-f579-4be2-9e4b-eb85cda9d88d.png">

![CleanShot 2022-10-03 at 10 45 08](https://user-images.githubusercontent.com/215660/193536643-aea8f7cf-0c10-45a1-bd8c-b69476dcfa5f.jpg)

Reply:
:eyes:

Reply:
Je to bug v libke:

Pridali tam `adaptErrorNamer`, ktory riesi bod `2.`:
![image](https://user-images.githubusercontent.com/19371734/193545770-8af15171-5a29-47f1-a10f-40a2600eb55f.png)

- Na viacerych miestach pouzivaju `errors.As`, ktore prechadza nadradene chyby a hlada dany typ.
- Ale `adaptErrorName` neobsahuje `Unwrap` metodu, tj. nefunguje s `errors.As`.
- Ked som tam "dolepil" `unwrap` tak to preslo spravne.
- Ono tam bude asi aj nejaka logicka chyba, ... kedze je to `ErrorNamer` (tj. iba nazov chyby), tak je logicke, ze nema `Unwrap` metodu, ... ale potom neviem, preco to pouzivaju s `errors.As`.


![image](https://user-images.githubusercontent.com/19371734/193545600-c4a8058d-e384-43e9-83e2-6d5590ef4fc2.png)

A preto to spadne, napr. na tomto mieste:
![image](https://user-images.githubusercontent.com/19371734/193546606-3e30802b-1ea0-4cac-be0d-00d1a364a90e.png)

-----

Skusim im spisat issue (asi az cez offsite), dame to zatial `on-hold`? 
Alebo staci aktualizacia na `3.8`?

Reply:
oomg. Díky! Tak na to nejsem z PHP resp. ani z Node.js zvyklej hledat bugy v importovaných libkách. 😅 Nechám to zatím na 3.8 a na 3.9 upgradneme později.